### ISSUE TITLE
JSON Schema Update

### ISSUE THREAD

**Issue by rmspeers** [spdx/spdx-spec#484]
The v2.2 JSON schema file hosted here does not seem to align to the v2.2 docs.

Specifically items like licenseDeclared that are marked as mandatory are not in the schema file. Also, mandatory fields are not marked mandatory.

It would be very helpful to have an up-to-date JSON schema to validate output from tools against to ensure compliance.

e.g., to packages would expect to see:
```
                "licenseDeclared" : {
                  "description" : "Provides a detailed description of the package.",
                  "type" : "string"
                }
```

I looked at https://github.com/spdx/tools to use potentially for this, but it doesn't handle Verify of JSON format files it appears. Open to other suggestions.

**Comment by goneall** [comment]
@rmspeers Agreed.

There is a new set of SPDX Java Tools which support the JSON format: https://github.com/spdx/tools-java

The Java library repository (used by the tools) is https://github.com/spdx/Spdx-Java-Library - the README describes the general structure and design.

One more repo of interest is https://github.com/spdx/spdx-java-jackson-store which implements the JSON serialization/deserialization.

The Schema file is generated from an RDF/OWL document (https://github.com/spdx/spdx-spec/blob/development/v2.2.1/ontology/spdx-ontology.owl.xml).  The code for the generator is at https://github.com/spdx/tools-java/blob/master/src/main/java/org/spdx/tools/schema/OwlToJsonSchema.java

Most likely, the code to generate the JSON schema needs to be improved, but it is possible there is a problem in the OWL document.

**Comment by rmspeers** [comment]
Thanks for taking a look. At least the basic example of licenseDeclared looks like it's at least _present_ in the OWL (https://github.com/spdx/spdx-spec/blob/development/v2.2.1/ontology/spdx-ontology.owl.xml#L327).

I'll try the Java tools for validation, although generation of JSON we're probably going to avoid going to Java to do.

**Comment by sschuberth** [comment]
I believe this has been fixed by now on the [v2.2.2 branch](https://github.com/spdx/spdx-spec/blob/development/v2.2.2/schemas/spdx-schema.json):

https://github.com/spdx/spdx-spec/blob/298a1fdbdbe5aba0d81398b0c82d198d2054a1e2/schemas/spdx-schema.json#L359-L362
